src/App.tsx

```diff
@@ -1,35 +1,236 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useState } from 'react';
+import './App.css';
+import styled from 'styled-components';
+import SearchField from './components/SearchField/SearchField';
+import Button from './components/Buttons/Button';
+import LogoutButton from './components/Buttons/LogoutButton';
+import kebabIcon from './assets/kebab-icon.svg';
+import kebabIconBlue from './assets/kebab-icon-blue.svg';
+
+const AppContainer = styled.div`
+  flex-direction: column;
+  margin-top: 20px;
+`;
+
+const Header = styled.div`
+  flex-direction: column;
+  margin-bottom: 16px;
+  width: 100%;
+  align-items: flex-start;
+
+  @media (min-width: 600px) {
+    display: grid;
+    grid-template-columns: auto 1fr; /* Auto-sized column for TodoTitle, 1fr-sized column for spacing, and auto-sized column for LogoutButton */
+  }
+`;
+
+const TitleRow = styled.div`
+  display: flex;
+  align-items: flex-start; /* Align items at the start (left) */
+  margin-bottom: 8px;
+
+  @media (min-width: 600px) {
+    margin-bottom: 0;
+    grid-column: 1 / -1; /* Span across all columns */
+  }
+`;
+
+const TodoTitle = styled.h1`
+  color: #2F80ED;
+  font-family: 'Roboto', sans-serif;
+  margin-right: 16px;
+  margin-top: 0;
+  font-size: 24px;
+
+  @media (min-width: 600px) {
+    font-size: 32px;
+  }
+`;
+
+const SearchFieldWrapper = styled.div`
+  display: flex;
+
+  @media (min-width: 600px) {
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    width: 100%;
+
+    & > *:first-child {
+      margin-right: 12px; /* Add margin to the right of the first child (SearchField) */
+    }
+  }
+`;
+
+const TodoItemContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: 16px;
+  margin-bottom: 8px;
+  cursor: pointer;
+
+  &:not(.action-buttons-visible):hover {
+    background-color: #f0f0f0;
+  }
+`;
+
+const TodoText = styled.div`
+  flex-grow: 1;
+`;
+
+const KebabIconContainer = styled.div`
+  position: relative;
+  display: inline-block;
+`;
+
+const KebabIcon = styled.img`
+  cursor: pointer;
+  transition: filter 0.3s;
+  filter: brightness(${(props) => (props.isVisible ? 1.5 : 1)});
+
+  ${KebabIconContainer}:hover &,
+  ${KebabIconContainer}:focus &,
+  &.blue {
+    filter: brightness(1.5);
+    content: url(${kebabIconBlue});
+  }
+`;
```

Comment:
```suggestion
const KebabIcon = styled.img<{isVisible: boolean}>`
  cursor: pointer;
  transition: filter 0.3s;
  filter: brightness(${(props) => (props.isVisible ? 1.5 : 1)});

  ${KebabIconContainer}:hover &,
  ${KebabIconContainer}:focus &,
  &.blue {
    filter: brightness(1.5);
    content: url(${kebabIconBlue});
  }
`;
```

Reply:
Can be refactored to this to remove the linting errors and provide type safety in styled components :bow: